src/Microsoft.ML.Transforms/RffTransform.cs

```diff
@@ -417,12 +419,11 @@ private static Float[] Train(IHost host, ColInfo[] infos, Arguments args, IDataV
                 else
                 {
                     Float[] distances;
-
                     var sub = args.Column[iinfo].MatrixGenerator;
-                    if (!sub.IsGood())
+                    if (sub == null)
                         sub = args.MatrixGenerator;
-                    var info = ComponentCatalog.GetLoadableClassInfo(sub);
-                    bool gaussian = info != null && info.Type == typeof(GaussianFourierSampler);
+                    var matrixGenerator = sub.CreateComponent(host, 1);
```

Comment:
>1 [](start = 68, length = 1)

Is it current default value?
Can we somehow get actual default value from argument class?
It's really easy to forget change this line if we decide to update default value in matrix generator.

Reply:
This is a "dummy" instance in order to get the type of matrix generator we are using.  See [https://github.com/dotnet/machinelearning/issues/699](https://github.com/dotnet/machinelearning/issues/699).

I'll add a comment.

---
In reply to: [211694368](https://github.com/dotnet/machinelearning/pull/700#discussion_r211694368) [](ancestors = 211694368)